### ISSUE TITLE
XrdCl metalink :: another case of no fallback to replica 

### ISSUE THREAD

**Issue by adriansev** [xrootd/xrootd#2210]
Hi @amadio ! I have another case of an EOS that returns an internal error and thus stopping the metalink processing to continue to the next replica.
See in attached file the log for the failed fallback, the meta that fails (the initial order)
and a working meta where a working replica was put first 

[file removed]

In this case it's unclear to me why the fallback does not work? Any idea?
Thanks a lot!

**Comment by adriansev** [comment]
ping @amadio just a gentle remainder :) while this was a rare case of an EOS fst misbehaving, it would be nice and useful if "continue to next replica" works in all conditions :)
Thanks a lot!

**Comment by amadio** [comment]
Sorry, @adriansev, I will fix this for 5.7.0, I've just been busy with other fixes before. I will let you know when I push the fix for this to the `devel` branch.

**Comment by adriansev** [comment]
Sure, no worries @amadio !! Thanks a lot!!

**Comment by adriansev** [comment]
Hi @amadio We encountered another case of internal error (logs [here](https://asevcenc.web.cern.ch/asevcenc/xrd_failed_fallback/)  ) and it seems the same thing..
If you have some time could you double check if this is the same thing? from the logs it seems the following: 
1. CENR fst st-192-e1bf9e17 is tried and answer with kXR_error
2. the metalink redirects to the following replica on ccxrdralice.in2p3.fr
3. some dialogue happens (i do not get who is the redirector and who is the data server at it seems that redirection to data server does not take place)
4. Open returns [3000] Internal error

Did i get it right? Is there anything more that can be done (beside your 648504f above) to make things (fallback) more robust?
Thanks a lot!!

**Comment by amadio** [comment]
I tried your metalink file and everything is working, so cannot quite reproduce the problem from before, but looks like the commit above should address it (although I did see some invalid argument errors in the logs, so not totally sure). The actual problem might have been some temporary issue on EOS ALICE.

I would recommend to not link such metalink files on GitHub due to security concerns. I took note of the address and removed the link from the comment above. Cheers,
